Source/WebKit/Shared/WebFoundTextRange.h

```diff
@@ -55,10 +55,9 @@ struct WebFoundTextRangeHash {
 
 template<> struct HashTraits<WebKit::WebFoundTextRange> : GenericHashTraits<WebKit::WebFoundTextRange> {
     static WebKit::WebFoundTextRange emptyValue() { return { }; }
-    static WebKit::WebFoundTextRange deletedValue() { return { std::numeric_limits<uint64_t>::max(), 0, { }, 0 }; }
 
-    static void constructDeletedValue(WebKit::WebFoundTextRange& range) { range = deletedValue(); }
-    static bool isDeletedValue(const WebKit::WebFoundTextRange& range) { return range == deletedValue(); }
+    static void constructDeletedValue(WebKit::WebFoundTextRange& slot) { new (NotNull, &slot) WebKit::WebFoundTextRange { std::numeric_limits<uint64_t>::max(), 0, { HashTableDeletedValue }, 0 }; }
+    static bool isDeletedValue(const WebKit::WebFoundTextRange& range) { return range.length && range.location == std::numeric_limits<uint64_t>::max(); }
```

Comment:
Seems like we could just check location here, no need to also check length.

Reply:
Or we could check just the value of the `AtomString`.

Reply:
Updated to check the value of the `AtomString`.